legate/core/operation.py

```diff
@@ -195,14 +209,29 @@ def add_dtype_arg(self, dtype):
     def add_future(self, future):
         self._futures.append(future)
 
+    def add_fusion_metadata(self, fusion_metadata):
+        self._is_fused = True
+        self._fusion_metadata = fusion_metadata
+
     def launch(self, strategy):
         launcher = TaskLauncher(self.context, self._task_id, self.mapper_id)
 
-        for input, input_part in zip(self._inputs, self._input_parts):
+        if self._is_fused:
+            launcher.add_fusion_metadata(self._is_fused, self._fusion_metadata)
+        if  self._is_fused: #fused ops re-use encapsulated unfused partitions
+            input_parts = self._unfused_input_parts
+            output_parts = self._unfused_output_parts
+            reduction_parts = self._unfused_reduction_parts
+        else:
+            input_parts = self._input_parts
+            output_parts = self._output_parts
+            reduction_parts = self._reduction_parts
+
```

Comment:
There's two way of generating the set of partitions for a fused op:
-1 is to create the fused op, then repartition all the stores of the fused op (even though they were already partitioned in there unfused form)
-The other way is to build a `superstrategy` of all the individual 'unfused' partitions
The second is much cheaper (partitioning_stores twice has significant/tangible overhead) and is what is currently used in the code